demos/smart_classroom_demo/main.cpp

```diff
@@ -573,56 +668,53 @@ int main(int argc, char* argv[]) {
         face_config.increase_scale_y = static_cast<float>(FLAGS_exp_r_fd);
         detection::FaceDetection face_detector(face_config);
 
-        // Load face detector for face database registration
-        detection::DetectorConfig face_registration_det_config(fd_model_path, fd_weights_path);
-        face_registration_det_config.deviceName = FLAGS_d_fd;
-        face_registration_det_config.ie = ie;
-        face_registration_det_config.enabled = !fd_model_path.empty();
-        face_registration_det_config.is_async = false;
-        face_registration_det_config.confidence_threshold = static_cast<float>(FLAGS_t_reg_fd);
-        face_registration_det_config.increase_scale_x = static_cast<float>(FLAGS_exp_r_fd);
-        face_registration_det_config.increase_scale_y = static_cast<float>(FLAGS_exp_r_fd);
-        detection::FaceDetection face_detector_for_registration(face_registration_det_config);
-
-        // Load face reid
-        CnnConfig reid_config(fr_model_path, fr_weights_path);
-        reid_config.enabled = face_config.enabled && !fr_model_path.empty() && !lm_model_path.empty();
-        reid_config.deviceName = FLAGS_d_reid;
-        if (checkDynamicBatchSupport(ie, FLAGS_d_reid))
-            reid_config.max_batch_size = 16;
-        else
-            reid_config.max_batch_size = 1;
-        reid_config.ie = ie;
-        VectorCNN face_reid(reid_config);
-
-        // Load landmarks detector
-        CnnConfig landmarks_config(lm_model_path, lm_weights_path);
-        landmarks_config.max_batch_size = 16;
-        landmarks_config.enabled = face_config.enabled && reid_config.enabled && !lm_model_path.empty();
-        landmarks_config.deviceName = FLAGS_d_lm;
-        if (checkDynamicBatchSupport(ie, FLAGS_d_lm))
+        std::unique_ptr<FaceRecognizer> face_recognizer;
+
+        if (face_config.enabled && !fr_model_path.empty() && !lm_model_path.empty()) {
+            // Create face tracker
+
+            detection::DetectorConfig face_registration_det_config(fd_model_path, fd_weights_path);
+            face_registration_det_config.deviceName = FLAGS_d_fd;
+            face_registration_det_config.ie = ie;
+            face_registration_det_config.is_async = false;
+            face_registration_det_config.confidence_threshold = static_cast<float>(FLAGS_t_reg_fd);
+            face_registration_det_config.increase_scale_x = static_cast<float>(FLAGS_exp_r_fd);
+            face_registration_det_config.increase_scale_y = static_cast<float>(FLAGS_exp_r_fd);
+
+            CnnConfig reid_config(fr_model_path, fr_weights_path);
+            reid_config.deviceName = FLAGS_d_reid;
+            if (checkDynamicBatchSupport(ie, FLAGS_d_reid))
+                reid_config.max_batch_size = 16;
+            else
+                reid_config.max_batch_size = 1;
+            reid_config.ie = ie;
+
+            CnnConfig landmarks_config(lm_model_path, lm_weights_path);
             landmarks_config.max_batch_size = 16;
```

Comment:
```suggestion

```

Reply:
This was in the original code, but sure, might as well get rid of it. Removed.